utils/pack.ts

```diff
@@ -0,0 +1,52 @@
+import execa from 'execa';
+import path from 'path';
+import fs from 'fs-extra';
+import { TurboDryRun } from './types';
+
+const rootDir = path.join(__dirname, '..');
+
+async function main() {
+  const { stdout: sha } = await execa('git', ['rev-parse', '--short', 'HEAD'], {
+    cwd: rootDir,
+  });
+  const { stdout: turboStdout } = await execa(
+    'turbo',
+    ['run', 'build', '--dry=json'],
+    {
+      cwd: rootDir,
+    }
+  );
+  const turboJson: TurboDryRun = JSON.parse(turboStdout);
+  for (const task of turboJson.tasks) {
+    const dir = path.join(rootDir, task.directory);
+    const packageJsonPath = path.join(dir, 'package.json');
+    const originalPackageObj = await fs.readJson(packageJsonPath);
+    const packageObj = await fs.readJson(packageJsonPath);
+    packageObj.version += `-${sha.trim()}`;
+
+    if (task.dependencies.length > 0) {
+      for (const dependency of task.dependencies) {
+        const name = dependency.split('#')[0];
+        const tarballUrl = `https://${process.env.VERCEL_URL}/tarballs/${name}.tgz`;
+        if (packageObj.dependencies && name in packageObj.dependencies) {
+          packageObj.dependencies[name] = tarballUrl;
+        }
+        if (packageObj.devDependencies && name in packageObj.devDependencies) {
+          packageObj.devDependencies[name] = tarballUrl;
+        }
+      }
+    }
```

Comment:
Can we extract the logic that produces `packageObj` and write some tests for it?